src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/EditorInProcess.cs

```diff
@@ -687,6 +687,22 @@ public async Task DismissCompletionSessionsAsync(CancellationToken cancellationT
         broker.DismissAllSessions(view);
     }
 
+    public async Task WaitForCompletionSessionsAsync(CancellationToken cancellationToken)
```

Comment:
@sharwell After `WaitForCompletionSetAsync` the `broker.IsCompletionActive(view)` returns `true`, while the completion `view` in the IDE might not shown yet and `broker.GetSessions(view)` returns an empty collection. Since I was hitting this pretty much 100% of times locally I added this helper. The behavior is either related to some async completion stuff or is a platform bug. Can you PTAL and, if needed, create an internal tracking issue for this? Also, shouldn't `WaitForCompletionSetAsync` correctly wait for the session to start in the first place?

Reply:
Also I didn't add calls to this helper in other completion-related tests, bug I suspect they are susceptible to this behavior and may be flaky due to this thing as well.

Reply:
The code which connects the editor behavior to `FeatureAttribute.CompletionSet` is here:

https://github.com/dotnet/roslyn/blob/55a809f91cae01bf1e6c635ee91faae6b0af23fe/src/VisualStudio/IntegrationTest/TestSetup/AsyncCompletionTracker.cs

The call to `GetSelectedItemAsync` is _intended_ to serve this purpose, but apparently is not doing so. That call will need to be updated to correctly invoke the operation that ensures the view is rendered.

Reply:
Can you please clarify what changes do we need to make? Or is it a platform issue (`ISelectedItemProvider` is a VS type)?